### PR TITLE
Orion/map collection with lookups

### PR DESCRIPTION
**Pull request by borland**

### CODE REVIEW

File: pkg/cmd/release/list/list.go [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+				return err
+			}
+
+			type ReleaseOutput struct {
+				Channel string
+				Project string
+				Version string
+			}
+
+			// page-fetching loop. TODO sync with dom
```
Comment (domenicsim1) [inline pkg/cmd/release/list/list.go:38]: Still want to sync?

File: pkg/cmd/release/list/list.go [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+			type ReleaseViewModel struct {
+				Channel   string
+				ChannelID string `json:",omitempty"`
+				Project   string
+				ProjectID string `json:",omitempty"`
+				Version   string
+			}
+
+			// page-fetching loop. TODO sync with dom
```
Comment (domenicsim1) [inline pkg/cmd/release/list/list.go:38]: Still want to sync?

File: pkg/util/util.go [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+func SliceContains[T comparable](slice []T, target T) bool {
+	for _, item := range slice {
+		if item == target {
+			return true
+		}
+	}
+	return false
+}
+
+func MapSlice[T any, TResult any](slice []T, mapper func(item T) TResult) []TResult {
```
Comment (borland) [inline pkg/util/util.go:12]: This is awkward. Every language other than C# calls this operation "map", however golang calls dictionaries "map" as well, so there's a confusion of terms.

Maybe we should rename it to `SliceSelect` to match C# or something? Perhaps `TransformSlice` to align with `std::transform` in C++ instead?. Or we could just leave it 🤷

Comment (domenicsim1) [reply pkg/util/util.go:12]: Let’s use SliceTransform

**Review by domenicsim1 — APPROVED** [review]
😎

File: pkg/util/util.go (hunk 3 above)
Comment (borland) [reply pkg/util/util.go]: Done :-)

File: pkg/cmd/release/list/list.go (hunk 2 above)
Comment (borland) [reply pkg/cmd/release/list/list.go:38]: I'll merge to main first and sync as part of create-release; that part is lower priority

**Pull request merged**